packages/ui-navigation/src/AppNav/index.tsx

```diff
@@ -187,32 +189,31 @@ class AppNav extends Component<AppNavProps> {
       </li>
     )
   }
-
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'items' implicitly has an 'any' type.
-  renderMenu(items) {
+  renderMenu(items: (ReactChild | ReactFragment | ReactPortal | Item)[]) {
     const menu = (
       <Menu
-        // @ts-expect-error ts-migrate(2339) FIXME: Property 'handleMenuDismiss' does not exist on typ... Remove this comment to see the full error message
-        onDismiss={this.handleMenuDismiss} // TODO: remove when INSTUI-2262 is resolved
         trigger={
           <AppNav.Item
             renderLabel={callRenderProp(this.props.renderTruncateLabel)}
           />
         }
       >
-        {/* @ts-expect-error ts-migrate(7006) FIXME: Parameter 'item' implicitly has an 'any' type. */}
         {items.map((item, index) => {
           return (
             <Menu.Item
-              href={item.props.href ? item.props.href : undefined}
+              href={
+                (item as Item).props.href
+                  ? (item as Item).props.href
+                  : undefined
+              }
               onClick={
-                item.props.onClick && !item.props.href
-                  ? item.props.onClick
-                  : null
+                (item as Item).props.onClick && !(item as Item).props.href
+                  ? (item as Item).props.onClick
+                  : () => {}
               }
               key={index}
             >
-              {callRenderProp(item.props.renderLabel)}
+              {callRenderProp((item as Item).props.renderLabel)}
```

Comment:
I'd make a new variable instead of using so many casts:
```
          const appNavItem = item as Item
          return (
            <Menu.Item
              href={
                appNavItem.props.href ? appNavItem.props.href : undefined
              }
              onClick={
                appNavItem.props.onClick && !appNavItem.props.href
                  ? appNavItem.props.onClick
                  : () => {}
              }
              key={index}
            >
              {callRenderProp(appNavItem.props.renderLabel)}
            </Menu.Item>
          )
```

Reply:
If it has the correct type (I commented on this too) then the casts are not needed at all